### DISCUSSION TITLE
Why not render to the document.body?

### DISCUSSION THREAD

**Discussion by thednp** [solidjs/solid-start#1152] [Q&A]
With the advent of new tech comes new opportunities, so I'm just wondering, if all gets modernized and re-invented, why not make the DOM faster by rendering/mounting directly into the `document.body`. Why not FIX this, IMO, broken pattern for good.

What I mean is: instead of having

```jsx
<!-- the current output markup -->
<html>
  <head />
  <body>
    <div id="root">
      <header />
      <main />
      <footer />
    </div>
  </body>
</html>
```
with mounting directive `render(App, document.getElementById('root'))`...

We should have

```jsx
<!-- the desired output markup -->
<html>
  <head />
  <body>
    <header />
    <main />
    <footer />
  </body>
</html>
```
with mounting directive `render(App, document.body)`.

Wouldn't this make more sense for a faster and more modern things to come with say... SolidStart or Vite examples?

PS: I've asked you Ryan during your last stream but you ignored it, probably for good reason.

**Comment by ryansolid** [comment]
It honestly doesn't matter a ton except it reduces diffing top level. If we render to body we end up diffing the other script tags and things added to the body and because Solid doesn't know about them we end up possibly removing them. This was just ensure things were cleaner. I believe current start uses `document.body` but I changed it because I was I think getting annoyed at hydration scripts disappearing.

**Comment by thednp** [comment]
Thanks @ryansolid for your reply. Indeed I can confirm the scripts get removed even if you're doing something like this

```tsx
<body>
  {children}
</body>
<div id="scripts">
  {scripts}
</div>
```


..but I wonder wouldn't this idea raise @nksaraf 's eye brow a little bit?
